ci/build_cpp.sh

```diff
@@ -13,7 +13,11 @@ export CMAKE_GENERATOR=Ninja
 
 rapids-print-env
 
-version=$(rapids-generate-version)
+# TODO: revert this once we start publishing nightly packages
+#       from the 'cugraph-gnn' repo and stop publishing them from
+#       the 'cugraph' / 'wholegraph' repos
+#version=$(rapids-generate-version)
+version="24.12.00a1000"
```

Comment:
This (and other hard-coded `24.12.00a1000` is the weirdest part of this PR.

They're here to ensure that testing CI jobs are getting the packages built from this repo. Because there have been more 24.12 commits over in https://github.com/rapidsai/cugraph, I observed test jobs pulling in nightlies from `rapidsai-nightly` instead of the ones downloaded from S3, which was a cause of errors and accidental successes.

Setting this ridiculous number is safe because no packages are being published to any conda channels or the nightly PyPI index from this repo yet.